### ISSUE TITLE
Tilt not working... again :)

### ISSUE THREAD

**Issue by vanosg** [thorrak/fermentrack#468]
Started a new ferment. I tilted the Tilt prior to dropping it in the bucket and saw the web UI pull a value. Now, 2 hours in, the values (both temp and gravity) have not changed. I used the phone app, and it is receiving beacons just fine (with what look to be accurate temp values). How do I troubleshoot what is going on here? The phone app shows the beacons coming in around ~73dbm strength.

Viewing the Tilt Debug on the help page, all the versions are listed as 'OK'.

tl;dr; Tilt values aren't being updated in the web UI for me.

**Comment by vanosg** [comment]
@thorrak any thoughts on this?

**Comment by thorrak** [comment]
Are you on master or dev?

**Comment by vanosg** [comment]
master

**Comment by thorrak** [comment]
Got it. All the fixes are currently stuck on dev, unfortunately. 

That said, if you’re in the middle of a ferment you might want to wait to upgrade - the upgrade also switches to Django 3.0 which has issues with Raspbian Stretch (requiring an upgrade to Python 3.7) and older databases due to the fact that Django 1.11 didn’t actually enforce foreign key restrictions on SQLite. Fun times. :(

**Comment by vanosg** [comment]
Ugh.. I thought I saw those fixes had been pushed to master, otherwise I'd be on dev! OK, thanks!

**Comment by vanosg** [comment]
Oh - if you haven't already, I'd request a 'last beacon seen' field somewhere, (and fix signal strength, which I think is still an open issue) so people can tell if they are getting current beacons... that would be a big help for me at least

**Comment by thorrak** [comment]
Signal strength I _think_ is fixed on dev. Last beacon seen isn’t quite there but can be added.

**Comment by vanosg** [comment]
Any progress on merging this one? The killer issue I'm looking at here is not knowing if I can trust the value being presented in the UI as being 'accurate' or 'recent'. Thanks as always

**Comment by thorrak** [comment]
Well, I _thought_ I had this (the "last beacon seen" bit) done. This has been added to the `docker` branch and will be released when that branch gets merged to dev/master. (Potentially soon)

**Comment by vanosg** [comment]
And I'll beat the dead horse one more time as well- for me, the most misleading area that could confuse a user is the front dashboard that displays the last value of the Tilt. If I log on real quick to see the status of the fermentation, that's where I'm going to see the gravity. If I were to log on right now, it would still tell me my beer is currently at 1.026 (Mmm, barleywine), even though it has been out of the fermenter for two weeks. There is no indication that displayed value is not accurate, which could cause issues if a) the connection is interrupted or b) the batter dies, etc etc. That's the part I think would be most important to update/change. Thanks thanks!

**Comment by thorrak** [comment]
> And I'll beat the dead horse one more time as well- for me, the most misleading area that could confuse a user is the front dashboard that displays the last value of the Tilt. If I log on real quick to see the status of the fermentation, that's where I'm going to see the gravity. If I were to log on right now, it would still tell me my beer is currently at 1.026 (Mmm, barleywine), even though it has been out of the fermenter for two weeks. There is no indication that displayed value is not accurate, which could cause issues if a) the connection is interrupted or b) the batter dies, etc etc. That's the part I think would be most important to update/change. Thanks thanks!

That's fixed by the second commit above referencing this issue. ;) Tilts now "expire" from the dashboards within 5 minutes and iSpindel readings within 1 hour - but are maintained so that combined beer and gravity logs will not have large gaps in data. 

Not ideal, but the best option I could think of given how the combined logs are produced/read.
